### PR TITLE
Added OpenSUSE aarch64

### PR DESCRIPTION
**Pull request by e-minguez**

### CODE REVIEW

**Review by afbjorklund — DISMISSED** [review]

**Comment by e-minguez** [comment]
> Strange this image isn't mentioned on the regular download page
> 
> https://get.opensuse.org/leap/15.4/#download
> 
> ----

TBH I saw the link here https://en.opensuse.org/openSUSE:AArch64#Running_openSUSE_from_raw_image

> 
> Seems correct, boots OK

🎉

**Comment by AkihiroSuda** [comment]
@jandubois LGTY?

File: examples/opensuse.yaml [hunk 1]
```diff
@@ -3,7 +3,10 @@ images:
 # Hint: run `limactl prune` to invalidate the "Current" cache
 - location: https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-JeOS.x86_64-15.4-OpenStack-Cloud-Current.qcow2
   arch: "x86_64"
-# No aarch64 OpenStack build found :(
+  digest: "sha256:378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195"
+- location: https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2
+  arch: "aarch64"
+  digest: "sha256:79d63c9096e13fc5b615c4807ac845b3afe6bed4f4d71ae257f764d4ac2e11fc"
```
Comment (AkihiroSuda) [inline examples/opensuse.yaml:9]: The digest changes day by day, so it shouldn't be pinned

File: examples/opensuse.yaml [hunk 2]
```diff
@@ -3,7 +3,10 @@ images:
 # Hint: run `limactl prune` to invalidate the "Current" cache
 - location: https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-JeOS.x86_64-15.4-OpenStack-Cloud-Current.qcow2
   arch: "x86_64"
-# No aarch64 OpenStack build found :(
+  digest: "sha256:378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195"
+- location: https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2
```
Comment (AkihiroSuda) [inline examples/opensuse.yaml:7]: JeOS is deprecated https://github.com/openSUSE/get-o-o/pull/123

Comment (jandubois) [reply examples/opensuse.yaml:7]: Yeah, I've asked internally to get a Minimal arm64 build with cloud-init support, but I have not received any notifications related to that.

But I think that would be a Tumbleweed build, not Leap.

Given that the amd64 build is also JeOS, I think using the JeOS build for arm64 is fine. I can ask for clarification on what the status is.

Comment (AkihiroSuda) [inline examples/opensuse.yaml:7]: Can't download the file (from Japan)
```console
$ LANG=C wget https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2
--2023-04-06 09:00:49--  https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2
Resolving download.opensuse.org (download.opensuse.org)... 2001:67c:2178:8::13, 195.135.221.134
Connecting to download.opensuse.org (download.opensuse.org)|2001:67c:2178:8::13|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://mirrorcache-jp.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2 [following]
--2023-04-06 09:00:50--  https://mirrorcache-jp.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2
Resolving mirrorcache-jp.opensuse.org (mirrorcache-jp.opensuse.org)... 2400:8902::f03c:93ff:fec8:af18, 139.162.108.149
Connecting to mirrorcache-jp.opensuse.org (mirrorcache-jp.opensuse.org)|2400:8902::f03c:93ff:fec8:af18|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://mirror-jp.firstyear.id.au/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2 [following]
--2023-04-06 09:00:50--  https://mirror-jp.firstyear.id.au/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2
Resolving mirror-jp.firstyear.id.au (mirror-jp.firstyear.id.au)... 2400:8902::f03c:93ff:fec8:afa1, 139.162.108.7
Connecting to mirror-jp.firstyear.id.au (mirror-jp.firstyear.id.au)|2400:8902::f03c:93ff:fec8:afa1|:443... connected.
HTTP request sent, awaiting response... 500 Internal Server Error
2023-04-06 09:00:51 ERROR 500: Internal Server Error.
```

Comment (jandubois) [reply examples/opensuse.yaml:7]: I suspect this is a server issue on that particular mirror. Maybe try a different one from https://mirrors.opensuse.org ?

FWIW, I just downloaded the image (from Canada), and it seems to work fine on M1.

**Comment by vmoutoussamy** [comment]
Hi,

If you wish to use ARM images built by the Minimal-VM team (formerly known as the JeOS team) from SUSE, please use the following:

- https://download.opensuse.org/distribution/leap/15.5/appliances/openSUSE-Leap-15.5-Minimal-VM.aarch64-kvm.qcow2 
- https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-Minimal-VM.aarch64-kvm.qcow2

As a reminder the Minimal-VM team at SUSE built identical images for SLES and openSUSE Leap/TW, but the openSUSE ARM community build their own images with their own packages lists/requirements. The openSUSE ARM Community didn't want to drop the JeOS name to avoid confusion with the openSUSE/SLES JeOS images built by SUSE, thus the SUSE team decided to rename JeOS to Minimal-VM starting with 15.4.

So if you see any images for 15.4 and 15.5 that still has "JeOS" in the name, this means that you are using an images built by the openSUSE ARM Community but not the ones built by the Minimal-VM team from SUSE.

I hope this clarify and helps you decide which images to use.

Regards,

**Comment by afbjorklund** [comment]
> It's unfortunately hidden behind the "Alternative downloads" link.

Actually that was the one I was _trying_ to link, but it was confused:

Download: https://get.opensuse.org/leap/15.4/

Alternative downloads: https://get.opensuse.org/leap/15.4/#

It is missing from both of them, referring to yet another third page.

https://en.opensuse.org/Portal:Arm#Release_supported_platforms

I guess it is the Just Enough Operating System confusion from above ?

See https://en.wikipedia.org/wiki/Conway%27s_law

**Comment by Vogtinator** [comment]
> (no "Cloud image" for arm(64))
> It is missing from both of them, referring to yet another third page.
> https://en.opensuse.org/Portal:Arm#Release_supported_platforms
> I guess it is the Just Enough Operating System confusion from above ?

Basically, yes. IMO having a wiki page for the huge list of "community" JeOS images makes sense, but it's confusing that the Minimal-VM images are listed on the download page explicitly. Ideally it should also be explained what the differences are...

File: examples/opensuse.yaml (hunk 2 above)
Comment (AkihiroSuda) [reply examples/opensuse.yaml:7]: Filed https://bugzilla.opensuse.org/show_bug.cgi?id=1210240

**Comment by AkihiroSuda** [comment]
> * https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-Minimal-VM.aarch64-kvm.qcow2

`Minimal-VM.aarch64-kvm` image doesn't seem to boot: https://github.com/lima-vm/lima/pull/1194#issuecomment-1326943869

`openSUSE-Leap-15.4-Minimal-VM.x86_64-OpenStack-Cloud.qcow2` works though.

File: examples/opensuse.yaml (hunk 1 above)
Comment (AkihiroSuda) [reply examples/opensuse.yaml:9]: @e-minguez Please drop this digest, and I'm happy to merge this PR.
Migration to Minimal-VM can be discussed later.

**Comment by AkihiroSuda** [comment]
Please squash commits, and update the PR title

File: examples/opensuse.yaml [hunk 3]
```diff
@@ -3,7 +3,9 @@ images:
 # Hint: run `limactl prune` to invalidate the "Current" cache
 - location: https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-JeOS.x86_64-15.4-OpenStack-Cloud-Current.qcow2
   arch: "x86_64"
-# No aarch64 OpenStack build found :(
+  digest: "sha256:378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195"
```
Comment (AkihiroSuda) [inline examples/opensuse.yaml:6]: This digest has to be removed too

File: examples/opensuse.yaml (hunk 1 above)
Comment (jandubois) [reply examples/opensuse.yaml]: I don't see the digest changing; it just looks like the digest in the PR is wrong. The correct digest should be:

```
$ cat openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2.sha256
965691484a2033f95789050820921ab847d1827a375038831aa7a237718099a5 openSUSE-Leap-15.4-ARM-JeOS-efi.aarch64.qcow2
```

File: examples/opensuse.yaml (hunk 3 above)
Comment (jandubois) [reply examples/opensuse.yaml:6]: Why, it seems to be correct:

```console
$ cat openSUSE-Leap-15.4-JeOS.x86_64-OpenStack-Cloud.qcow2.sha256
378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195 openSUSE-Leap-15.4-JeOS.x86_64-OpenStack-Cloud.qcow2
```

And I've successfully created an openSUSE instance with it:

```console
$ l start examples/opensuse.yaml
? Creating an instance "opensuse" Proceed with the current configuration
INFO[0000] Attempting to download the image              arch=x86_64 digest="sha256:378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195" location="https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-JeOS.x86_64-15.4-OpenStack-Cloud-Current.qcow2"
INFO[0001] Using cache "/Users/jan/Library/Caches/lima/download/by-url-sha256/27f3bb70bbc178edcae8edd360883ca9f2c9dc88c9d83cecd7d5b9752347ad05/data"
[...]
```

Comment (jandubois) [reply examples/opensuse.yaml:6]: The JeOS checksum seems unchanged across some mirrors:

```console
$ FILE=distribution/leap/15.4/appliances/openSUSE-Leap-15.4-JeOS.x86_64-OpenStack-Cloud.qcow2.sha256
$ curl https://download.opensuse.org/$FILE
378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195 openSUSE-Leap-15.4-JeOS.x86_64-OpenStack-Cloud.qcow2
$ curl https://mirror.csclub.uwaterloo.ca/opensuse/$FILE
378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195 openSUSE-Leap-15.4-JeOS.x86_64-OpenStack-Cloud.qcow2
$ curl https:///mirror.its.dal.ca/opensuse/$FILE
378c51ae8a064513410285ffe884fa455e6325c583861aebe11cff7291298195 openSUSE-Leap-15.4-JeOS.x86_64-OpenStack-Cloud.qcow2
```

However, there is something wrong with the checksums on openSUSE mirrors; I've noticed this for the Minimal image now:

```console
$ FILE=distribution/leap/15.4/appliances/openSUSE-Leap-15.4-Minimal-VM.x86_64-OpenStack-Cloud.qcow2.sha256
$ curl https://download.opensuse.org/$FILE
c686339ace6ab5d3cdf6999d00b3dec2b8adf11456f322d970d6caef74614381 openSUSE-Leap-15.4-Minimal-VM.x86_64-OpenStack-Cloud.qcow2
$ curl https://mirror.csclub.uwaterloo.ca/opensuse/$FILE
07a770c25f26a1b298e862c6b24096f80e0e4c37938cffa7f9261854eba15e7a openSUSE-Leap-15.4-Minimal-VM.x86_64-OpenStack-Cloud.qcow2
$ curl https://mirror.its.dal.ca/opensuse/$FILE
07a770c25f26a1b298e862c6b24096f80e0e4c37938cffa7f9261854eba15e7a openSUSE-Leap-15.4-Minimal-VM.x86_64-OpenStack-Cloud.qcow2
```

So different mirrors have different images and different checksums:

```console
$ FILE=distribution/leap/15.4/appliances/openSUSE-Leap-15.4-Minimal-VM.x86_64-OpenStack-Cloud.qcow2
$ curl -so uwaterloo.qcow2 https://mirror.csclub.uwaterloo.ca/opensuse/$FILE
$ curl -so opensuse.qcow2 https://downloadcontent.opensuse.org./$FILE
$ sha256sum uwaterloo.qcow2
07a770c25f26a1b298e862c6b24096f80e0e4c37938cffa7f9261854eba15e7a  uwaterloo.qcow2
$ sha256sum opensuse.qcow2
c686339ace6ab5d3cdf6999d00b3dec2b8adf11456f322d970d6caef74614381  opensuse.qcow2
```

So at least the servers seem to be internally consistent, but I guess it means the images are being rebuilt with the same names, so these are not repeatable downloads and can't have a digest in the template.

**Comment by jandubois** [comment]
> If you wish to use ARM images built by the Minimal-VM team (formerly known as the JeOS team) from SUSE, please use the following:
> 
> * https://download.opensuse.org/distribution/leap/15.5/appliances/openSUSE-Leap-15.5-Minimal-VM.aarch64-kvm.qcow2
> * https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-Minimal-VM.aarch64-kvm.qcow2

@vmoutoussamy The "kvm" images lack support for cloud-init, so don't work for Lima. We need the "OpenStack-Cloud" flavour. So this image works:

```
https://download.opensuse.org/distribution/leap/15.4/appliances/openSUSE-Leap-15.4-Minimal-VM.x86_64-OpenStack-Cloud.qcow2
```

But just as for Tumbleweed, there is no corresponding aarch64 image.

I'm also somewhat surprised that the Leap 15.4 images seem to get rebuilt over time. I expected this for Tumbleweed, but for Leap I thought you would get a reproducible image and not a snapshot with an undetermined number of updates already applied.

I guess I can see arguments both ways, so I guess I just want to confirm that this is a rolling release, and checksums are meaningless unless downloaded from the same mirror at the exact same time.

**Comment by e-minguez** [comment]
> Please squash commits, and update the PR title

Sorry, I went AFK for a few days. It should be sorted out now. Thanks!

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**